lib/themes/dosomething/paraneue_dosomething/js/takeover/Takeover.js

```diff
@@ -0,0 +1,245 @@
+import Validation from 'dosomething-validation';
+import setting from '../utilities/Setting';
+
+// @IDEA - Let this just take care of screen interactions, pull in custom stuff via classes
+import takeoverTpl from 'takeover/templates/takeover.tpl.html';
+
+const $ = require('jquery');
+const template = require('lodash/string/template');
+
+const Takeover = {
+  baseUrl: $(location).attr("origin"),
+
+  spinner: '<div class="spinner"></div>',
+
+  screens: null,
+
+  firstScreen: null,
+
+  lastScreen: null,
+
+  screenIndexes: null,
+
+  /**
+   * Initialize the Signup interface.
+   *
+   * @param  {jQuery} $takeoverContainer the element to add the taked over to.
+   */
+  init: function($takeoverContainer = $('.takeover-container')) {
+    const data = {
+      'header': 'Thanks for signing up',
+    }
+
+    // Add the takeover to the page.
+    const slide = $takeoverContainer.hasClass('-no-slide') ? false : true;
+    this.addTakeover($takeoverContainer, data, slide);
+
+    // Prepare validation fields that got entered into the DOM after page load.
+    Validation.prepareFields($('input'));
+
+    // Set up interactions
+    this.screens = $('.takeover__screen');
+    this.screenIndexes = this.setScreenIndexes();
+    this.firstScreen = this.screens.first();
+    this.lastScreen = this.screens.last();
+    this.firstScreen.addClass('active');
+
+    // Set up submit handlers.
+    this.submitButton = $('.submit-postal-code');
+    this.handleSubmit(this.submitButton, {
+      'button' : this.submitButton,
+      'method' : 'GET',
+      'url'    : 'http://jsonplaceholder.typicode.com/posts/1',
+      'data'   : null,
+    });
+  },
+
+  /**
+   * Adds the takeover to the page with or without a slide animation.
+   *
+   * @param  {jQuery} $container  the container to add the takeover to.
+   * @param  {object} $data       An object containing the data to inject into the templates
+   * @param  {bool}   slide       Whether or not to add the takeover with a slide animation.
+   */
+  addTakeover: function($container, data, slide = false) {
+    const markup = template(takeoverTpl);
+    $container.html(markup(data));
+    this.takeoverContent = $('.takeover');
+
+    if (slide) {
+      this.takeoverContent.hide();
+      this.takeoverContent.delay(500).slideDown('slow', function() {});
+    }
+  },
+
+  /**
+   * Binds click handlers to $button that sends an AJAX request and uses deferred promises to
+   * run callbacks in the order in which they are recieved.
+   *
+   * @param  {jQuery}   $button   Signup buttons
+   * @param  {object}   request   An object that defines the properties we need to make a request.
+   */
+  handleSubmit: function($button, request) {
+    const _this = this;
+
+    // Create custom deferred promise object.
+    let ajaxPromise = $.Deferred().resolve().promise();
```

Comment:
Definitely not something needed for merge, but it'd be pretty cool to start using [fetch](https://jakearchibald.com/2015/thats-so-fetch/) via [the polyfill](https://github.com/github/fetch) since the way jQuery handles promises is so wonky.